File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) 
    {
        PriorityQueue<Long> pq=new PriorityQueue<>();//min heap

        for(int i=0;i<nums.length;i++)
        {
            pq.add((long)nums[i]);
        }

        int count=0;

        while(pq.size()>=2 && pq.peek()<k )
        {
            long x=pq.remove();
            long y=pq.remove();

            pq.add((long)(x*2+y));

            count++;

        }

        return count;    
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.kt
class Solution {
    fun minOperations(nums: IntArray, k: Int): Int {

        var pq=PriorityQueue<Long>()

        for(num in nums)
        {
            pq.add(num.toLong())
        }

        var count=0

        while(pq.size>=2 && pq.peek()<k)
        {
            val x=pq.remove()
            val y=pq.remove()

            pq.add(x*2+y)

            count++
        }

        return count
        
    }
}